core/metadata.cpp

```diff
@@ -0,0 +1,83 @@
+// SPDX-License-Identifier: GPL-2.0
+#include "metadata.h"
+#include "exif.h"
+#include "qthelper.h"
+#include <QString>
+#include <QFile>
+#include <QDateTime>
+
+// Fetch quint16 in big endian mode from QFile and return 0 on error.
+// This is a very specialized function for parsing JPEGs, therefore we can get away with such an in-band error code.
+static inline quint16 getShortBE(QFile &f)
+{
+	unsigned char buf[2];
+	if (f.read(reinterpret_cast<char *>(buf), 2) != 2)
+		return 0;
+	return (buf[0] << 8) | buf[1];
+}
+
+static int parseExif(const QString &filename, easyexif::EXIFInfo &exif)
+{
+	QFile f { filename };
```

Comment:
What's the difference between
`QFile f(filename);
`
and
`QFile f { filename };
`
?

(again, I realize that this is just copied code)

Reply:
These do the same thing.
I think the latter is preferred by some because of the "most vexing parse". I.e. it could also be a function declaration?